py/torch_tensorrt/dynamo/lowering/_decompositions.py

```diff
@@ -1,10 +1,182 @@
+    aten.im2col,
+    aten.index_add,
+    aten.index_add_,
+    aten.index_copy,
+    aten.index_copy_,
+    aten.index_fill,
+    aten.index_fill_,
+    aten.index_select,
+    aten.isneginf,
+    aten.isposinf,
+    aten.l1_loss,
+    aten.leaky_relu,
+    aten.leaky_relu_,
+    aten.leaky_relu_backward,
+    aten.lerp,
+    aten.linspace,
+    aten.logaddexp,
+    aten.logaddexp2,
+    aten.logit,
+    aten.logit_backward,
+    aten.log_sigmoid_backward,
+    aten.log_sigmoid_forward,
+    aten._log_softmax,
+    aten._log_softmax_backward_data,
+    aten.logspace,
+    aten.logsumexp.default,
+    aten.masked_fill,
+    aten.masked_fill_,
+    aten.max_pool2d_with_indices_backward,
+    aten.mish,
+    aten.mse_loss,
+    aten.mse_loss_backward,
+    aten.mv,
+    aten.mvlgamma,
+    aten.nansum,
+    aten.nan_to_num,
+    aten.narrow,
+    # TODO: Disable the below operators once freezing is done
+    aten.native_batch_norm,
+    aten.native_batch_norm_backward,
+    aten._native_batch_norm_legit,
+    aten._native_batch_norm_legit_functional,
+    aten._native_batch_norm_legit_no_training,
+    aten.native_dropout_backward,
+    aten.native_group_norm,
+    aten.native_group_norm_backward,
+    aten.native_layer_norm,
+    aten.native_layer_norm_backward,
+    aten.new_empty,
+    aten.new_full,
+    aten.new_ones,
+    aten.new_zeros,
+    aten.nll_loss_backward,
+    aten.nll_loss_forward,
+    aten.norm,
+    aten.ones,
+    aten.ones_like,
+    aten._prelu_kernel,
+    aten._prelu_kernel_backward,
+    aten._reshape_alias,
+    aten.rad2deg,
+    aten.renorm,
+    aten.renorm_,
+    aten.rot90,
+    aten.rsub.Scalar,
+    aten.rsub.Tensor,
+    aten.select_backward,
+    aten.select_scatter,
+    aten.sgn,
+    aten.sigmoid_backward,
+    aten.silu,
+    aten.silu_,
+    aten.silu_backward,
+    aten.sinc,
+    aten.slice_backward,
+    aten.smooth_l1_loss,
+    aten.smooth_l1_loss_backward,
+    aten.soft_margin_loss,
+    aten.soft_margin_loss_backward,
+    aten._softmax,
+    aten._softmax_backward_data,
+    aten.softplus,
+    aten.softplus_backward,
+    aten.softshrink,
+    aten.softshrink_backward,
+    aten.special_entr,
+    aten.special_log_ndtr,
+    aten.special_xlog1py,
+    aten.stack,
+    aten.t,
+    aten.tanh_backward,
+    aten.threshold,
+    aten.threshold_backward,
+    aten.trace,
+    aten.transpose.int,
+    aten.tril.default,
+    aten.triu.default,
+    aten.unfold,
+    aten.unfold_backward,
+    aten.unfold_copy,
+    aten.upsample_bilinear2d,
+    aten.upsample_bilinear2d.vec,
+    aten.upsample_nearest2d_backward,
+    aten.xlogy,
+    aten.zero,
+    aten.zero_,
+    aten.zeros,
+    aten.zeros_like,
+}
+disabled_decompositions: Set[torch._ops.OpOverload] = {}
 
-aten = torch.ops.aten
+ENABLED_TORCH_DECOMPOSITIONS: Dict[
+    torch._ops.OpOverload, Callable
+] = get_torch_decompositions(enabled_decompositions)
+TORCH_TRT_DECOMPOSITIONS: Dict[torch._ops.OpOverload, Callable] = {}
```

Comment:
The decompositions are three dictionaries:

- **`ENABLED_TORCH_DECOMPOSITIONS`** - the enabled decompositions we've pre-selected
- **`CORE_ATEN_DECOMPOSITIONS_FILTERED`** (defined in `get_decompositions` below) - the complete set of `_core_aten_decompositions` Torch provides, minus the set of disabled decompositions. Note that `TORCH_DECOMPOSITIONS` _may not_ be a subset of this set
- **`TORCH_TRT_DECOMPOSITIONS`** - the decompositions we've written ourselves

Reply:
Would TORCH_DECOMPOSITIONS only include decompositions from the get_torch_decompoistions set?

Reply:
`Note that TORCH_DECOMPOSITIONS may not be a subset of this set`
From what I understand, 
* it seems like some decompositions in `_core_aten_decompositions` have been removed eg: `aten.native_layer_norm`.  [ENABLED_TORCH_DECOMPOSITIONS](https://github.com/pytorch/TensorRT/blob/368a20e99d636b04723fd5e29bdd7d11084434e8/py/torch_tensorrt/dynamo/lowering/_decompositions.py#L14C1-L14C23) is a more complete set (from previous commit maybe ). Is this correct ? 
* In that case, what if we move aten.native.layer_norm to TORCH_TRT_DECOMPOSITIONS since it is useful to us and maybe other useful ones instead of maintaining a `ENABLED_TORCH_DECOMPOSITIONS` which overlaps with `_core_aten_decompositions` one ?

Reply:
@peri044 
- The interpretation of `ENABLED_TORCH_DECOMPOSITIONS` is correct
- My initial intent for `TORCH_TRT_DECOMPOSITIONS` was that it would only store decompositions we specifically (custom) wrote, not ones sourced from Torch, as `layer_norm` would be.